controllers/utils/base_reconciler.go

```diff
@@ -315,16 +323,19 @@ func (r *ReconciliationRunner) RunReconciliationStages(fns ...CloudStackReconcil
 // subject, get CAPI and CloudStackClusters, and call either r.Reconcile or r.ReconcileDelete.
 func (r *ReconciliationRunner) RunBaseReconciliationStages() (res ctrl.Result, retErr error) {
 	defer func() {
-		if err := r.Patcher.Patch(r.RequestCtx, r.ReconciliationSubject); err != nil {
-			if !strings.Contains(err.Error(), "is invalid: status.ready") {
-				err = errors.Wrapf(err, "error patching reconciliation subject")
-				retErr = multierror.Append(retErr, err)
+		if r.Patcher != nil {
```

Comment:
Turns out this could happen. 
It would happen if the reconciliation subject requested doesn't exist.
I think it's really rare that the request is for a reconciliation subject that's missing, but it's possible.